trlx/trainer/nn/ilql_models.py

```diff
@@ -62,12 +63,15 @@ class ILQLConfig(MethodConfig):
     def heads(self, hidden_size: int, vocab_size: int, dtype: type):
         return ILQLHeads(self, hidden_size, vocab_size, dtype)
 
-    def loss(self, outputs, labels: ILQLBatch):
+    def loss(self, outputs, labels):
         logits, (qs, target_qs, vs) = outputs
         terminal_mask = labels.dones[:, :-1]
         n_nonterminal = max(1, terminal_mask.sum())
-
-        actions = labels.input_ids[:, 1:].gather(dim=1, index=labels.actions_ixs).unsqueeze(-1)
+        # check type of labels
+        if isinstance(labels, ILQLBatch):
+            actions = labels.input_ids[:, 1:].gather(dim=1, index=labels.actions_ixs).unsqueeze(-1)
+        else:
+            actions = labels.decoder_input_ids[:, 1:].gather(dim=1, index=labels.actions_ixs).unsqueeze(-1)
```

Comment:
why do we want to `gather` here, wouldn't `decoder_input_ids` be already compiled in the needed format from `make_experience`

Reply:
yes, you are right we do no need the gather here. Actions also started with 0. Will remove it, but it not effect current result.